### ISSUE TITLE
Inductor: retain the default reduction config in max-autotune

### ISSUE THREAD

**Issue by IvanYashchuk** [pytorch/pytorch#194736]
### 🐛 Describe the bug

The max-autotune reduction set can omit the configuration chosen by the default persistent-reduction heuristic. More tuning can then exclude the best known candidate.
Max-autotune should start with the exact default candidate. It can then append its wider search set and remove duplicates.

## Motivation

FP16 row mean with 2,048 rows reproduces the gap at three reduction widths:

| Width | Default config | Default | Best max-only candidate | Advantage |
| ---: | --- | ---: | ---: | ---: |
| 256 | X4, one warp | 4.160 us | 4.288 us | 1.031x |
| 512 | X2, one warp | 5.504 us | 5.856 us | 1.064x |
| 1024 | X1, one warp | 4.832 us | 5.856 us | 1.212x |

Current source builds persistent-reduction configurations here:
[`triton_heuristics.py`](https://github.com/pytorch/pytorch/blob/b1716d913ad13513f6184bad3d025a5b259329f3/torch/_inductor/heuristics/triton_codegen/reduction.py#L416-L498).

## Reproducer

```python
import torch
import triton


def f(x):
    return x.mean(dim=-1)


for width in (256, 512, 1024):
    x = torch.randn(2048, width, device="cuda", dtype=torch.float16)
    for mode in (None, "max-autotune-no-cudagraphs"):
        compiled = torch.compile(f, fullgraph=True, mode=mode)
        torch.testing.assert_close(compiled(x), f(x))
        for _ in range(100):
            compiled(x)
        print(width, mode, triton.testing.do_bench(lambda: compiled(x)))
```

Use `TORCH_LOGS=output_code` to compare the selected launch configurations.

### Versions

Latest main (b1716d913ad13513f6184bad3d025a5b259329f3)

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo
